pymatgen/analysis/piezo_sensitivity.py

```diff
@@ -0,0 +1,602 @@
+    Phonopy = None
+
+__author__ = "Handong Ling"
+__copyright__ = "Copyright 2019, The Materials Project"
+__version__ = "1.0"
+__maintainer__ = "Handong Ling"
+__email__ = "hling@lbl.gov"
+__status__ = "Development"
+__date__ = "Feb, 2019"
+
+class BornEffectiveChargeTensor(Tensor):
+    """
+    This class describes the Nx3x3 born effective charge tensor
+    """
+
+    def __new__(cls, input_array, tol=1e-3):
+        """
+        Create an BornEffectiveChargeTensor object.  The constructor throws 
+        an error if the input_matrix argument does not satisfy the acoustic 
+        sum rule. Note that the constructor uses __new__ rather than __init__ 
+        according to the standard method of subclassing numpy ndarrays.
+
+        Args:
+            input_matrix (Nx3x3 array-like): the Nx3x63 array-like
+                representing the born effective charge tensor
+        """
+        obj = super(PiezoTensor, cls).__new__(cls, input_array, check_rank=4)
+        if not (np.sum(obj)) < tol).all():
+            warnings.warn("Input born effective charge tensor does "
+                          "not satisfy acoustic sum rule")
+        return obj.view(cls)
+
+class InternalStrainTensor(Tensor):
+    """
+    This class describes the Nx3x3x3 born effective charge tensor
+    """
+
+    def __new__(cls, input_array, tol=1e-3):
+        """
+        Create an InternalStrainTensor object.  The constructor throws an error if
+        the shape of the input_matrix argument is not Nx3x3x3, i. e. in true
+        tensor notation. Note that the constructor uses __new__ rather than
+        __init__ according to the standard method of subclassing numpy
+        ndarrays.
+
+        Args:
+            input_matrix (Nx3x3x3 array-like): the Nx3x3x3 array-like
+                representing the internal strain tensor
+        """
+        obj = super(PiezoTensor, cls).__new__(cls, input_array, check_rank=4)
+        if not (obj - np.transpose(obj, (0, 1, 3, 2)) < tol).all():
+            warnings.warn("Input internal strain tensor does "
+                          "not satisfy standard symmetries")
+        return obj.view(cls)
+
+class ForceConstantMatrix(Tensor):
+    """
+    This class describes the NxNx3x3 born effective charge tensor
+    """
+
+    def __new__(cls, input_array, tol=1e-3):
+        """
+        Create an ForceConstantMatrix object.  The constructor throws an error if
+        the shape of the input_matrix argument is not NxNx3x3, i. e. in true
+        tensor notation. Note that the constructor uses __new__ rather than
+        __init__ according to the standard method of subclassing numpy
+        ndarrays.
+
+        Args:
+            input_matrix (NxNx3x3 array-like): the NxNx3x3 array-like
+                representing the force constant matrix
+        """
+        obj = super(PiezoTensor, cls).__new__(cls, input_array, check_rank=4)
+
+        return obj.view(cls)
+
+
+def get_rand_BEC(struc, eqop, symops, max_charge=1):
```

Comment:
These functions should be methods of the corresponding Tensor class. Symops should probably default to None, and if so then you use SpaceGroupAnalyzer to find the symops. If not, you use the given symops.